run.php

```diff
@@ -76,40 +76,73 @@
+$source_formats = getSourceFormats("en");
+$target_formats = getTargetFormats("en");
 
-if ($source_formats->num_rows > 0) {
-    // output data of each row
-    while ( $r = $source_formats->fetch_assoc() ) {
-        $formatId = formatExists($target_formats, $r);
-        $notes = "exact match";
-        if ($formatId == 0) {
-            $notes = "unique format";
-            $formatId = $r["shared_id_bigint"] + $formats_max_id;
-            $insert_sql = "INSERT INTO " . $target_table_prefix . "comdef_" . $table_suffix . " VALUES (" .
-                          ($r["shared_id_bigint"] + $formats_max_id) . "," .
-                          "'" . $r["key_string"] . "'," .
-                          "NULL," .
-                          "'" . $r["worldid_mixed"] . "'," .
-                          "'" . $r["lang_enum"] . "'," .
-                          "'" . $GLOBALS['target_conn']->escape_string($r["name_string"]) . "'," .
-                          "'" . $GLOBALS['target_conn']->escape_string($r["description_string"]) . "'," .
-                          "'" . $r["format_type_enum"] . "')";
-            $insert_result = executeTargetDbQuery($insert_sql);
+$oldFormatIdToNewFormatId = array();
+
+if (count($source_formats) > 0) {
+    foreach ($source_formats as $source_format) {
+        if (!anySourceMeetingsUsingFormat($source_format->shared_id_bigint)) {
+            echo "Skipping source format '" . $source_format->key_string . ":" . $source_format->shared_id_bigint . "' because it is not in use.\n";
+            continue;
+        }
+        $found_match = false;
+        if (array_key_exists($source_format->shared_id_bigint, $oldFormatIdToNewFormatId)) {
+            $existingTargetFormatId = $source_formats[$source_format->shared_id_bigint];
+            continue;
+        }
+
+        // Look for exact matches
+        foreach ($target_formats as $target_format) {
+            if ($source_format->is_exact_match($target_format)) {
+                echo "Found exact format match: source format '" . $source_format->key_string . ":" . $source_format->shared_id_bigint . "' to target format '" . $target_format->key_string . ":" . $target_format->shared_id_bigint . "'.\n";
+                $oldFormatIdToNewFormatId[$source_format->shared_id_bigint] = $target_format->shared_id_bigint;
+                $found_match = true;
+                break;
+            }
+        }
+        if ($found_match) {
+            continue;
         }
 
-        if ($r["lang_enum"] == "en") {
-            insertIntoMergeTable($table_suffix, $r['shared_id_bigint'], $formatId, $notes);
+        // No exact matches, look for partial match
+        // If the source format has a format_type_enum, description_string, or worldid_mixed set and the target does not,
+        // we update the target with the source's values.
+        foreach ($target_formats as $target_format) {
+            if ($source_format->is_match($target_format)) {
+                /*
```

Comment:
The intent of this commented out section is to bring over the `format_type_enum`, `description_string`, and `worldid_mixed` values from the source if they do not exist in the target. For simplicity, I opted not do that yet, so it's commented.